simple/testdata/LintStringCopy.go

```diff
@@ -0,0 +1,15 @@
+package pkg
+
+func fn(s string) {
+	_ = string([]byte(s)) // MATCH /redundant string copy/
+	_ = "" + s // MATCH /redundant string copy/
+	_ = s + "" // MATCH /redundant string copy/
+
+	_ = s
+	_ = s + "foo"
+
+	string := func(v interface{}) string {
+		return "foo"
+	}
+	_ = string([]byte(s))
+}
```

Comment:
We'll want at least two more tests. One for `string([]rune(s))` and one for `type byte rune; string([]byte(s))`, neither of which should match.